lib/capitano.js

```diff
@@ -0,0 +1,188 @@
+        value: build.commit_hash
+      }
+    })
+    choices.splice(0, 0, {'name': LATEST, 'value': LATEST})
+    return form.ask({
+      message: 'Select a build',
+      type: 'list',
+      default: LATEST,
+      choices: choices
+    })
+  })
+}
+
+function disableAutomaticUpdates (form, resin, application, commit) {
+  if ((commit === LATEST) || !application.should_track_latest_release) {
+    return Promise.resolve()
+  }
+  const message = (
+    'This application is set to automatically update all devices to the ' +
+    'latest available version.\n' +
+    'You need to disable this if you want to preload a specific build. ' +
+    'Do you want to disable automatic updates for this application?'
+  )
+  return form.ask({
+    message: message,
+    type: 'list',
+    default: true,
+    choices: [
+      {name: 'Yes', value: true},
+      {name: 'No', value: false}
+    ]
+  })
+  .then(function (update) {
+    if (!update) {
+      return
+    }
+    return resin.pine.patch({
+      resource: 'application',
+      id: application.id,
+      body: {
+        should_track_latest_release: false
+      }
+    })
+  })
+}
+
+module.exports = {
+  signature: 'preload <image>',
+  description: '(beta) preload an app on a disk image',
+  help: `Warning: 'resin preload' requires Docker to be correctly installed in
+your shell environment. For more information (including Windows support)
+please check the README here: https://github.com/resin-io/resin-cli .
+
+Use this command to preload an application to a local disk image.
+
+Examples:
+  $ resin preload resin.img --app 12345 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image some-image.png
+  $ resin preload resin.img`,
+  permission: 'user',
+  primary: true,
+  options: [
+    {
+      signature: 'app',
+      parameter: 'appId',
+      description: 'id of the application to preload',
+      alias: 'a'
+    },
+    {
+      signature: 'commit',
+      parameter: 'hash',
+      description: 'a specific application commit to preload (ignored if no appId is given)',
+      alias: 'c'
+    },
+    {
+      signature: 'splash-image',
+      parameter: 'splashImage.png',
+      description: 'path to a png image to replace the splash screen',
+      alias: 's'
+    },
+    {
+      signature: 'dont-detect-flasher-type-images',
+      boolean: true,
+      description: 'Disables the flasher type images detection: treats all images as non flasher types'
+    }
+  ],
+  action: function (params, options, done) {
+    console.log(options)
+    const resin = require('resin-sdk-preconfigured')
+    const form = require('resin-cli-form')
+
+    options.image = params.image
+    options.appId = options.app
+    delete options.app
+
+    options.dontDetectFlasherTypeImages = options['dont-detect-flasher-type-images']
+    delete options['dont-detect-flasher-type-images']
+
+    return preload.build()
+    .then(resin.settings.getAll)
```

Comment:
> I have to disagree here, the caller doesn't need a stream, it just needs a Promise of when the build will be done.

This module is installable by other consumers than our CLI, and the caller should receive a stream of the child process' data, instead of it writing out to stdout, so it can handle & process the data.